enhancements/sig-architecture/136-placement-cel-selector/README.md

```diff
@@ -0,0 +1,343 @@
+- Variable `managedCluster` will be added.
+- Function `scores` will be added. 
+- Function `versionIsGreaterThan`, `versionIsLessThan` will be added. 
+- Function `quantityIsGreaterThan`, `quantityIsLessThan` will be added. 
+
+```go
+// ManagedClusterLib defines the CEL library for ManagedCluster evaluation.
+// It provides functions and variables for evaluating ManagedCluster properties
+// and their associated resources.
+//
+// Variables:
+//
+// managedCluster
+//
+// Provides access to ManagedCluster properties.
+//
+// Functions:
+//
+// scores
+//
+// Returns a list of AddOnPlacementScoreItem for a given cluster and AddOnPlacementScore resource name.
+//
+//	scores(<ManagedCluster>, <string>) <list>
+//
+// The returned list contains maps with the following structure:
+//   - name: string - The name of the score
+//   - value: int - The numeric score value
+//   - quantity: number|string - The quantity value, represented as:
+//     * number: for pure decimal values (e.g., 3)
+//     * string: for values with units or decimal places (e.g., "300Mi", "1.5Gi")
+//
+// Examples:
+//
+//	managedCluster.scores("cpu-memory") // returns [{name: "cpu", value: 3, quantity: 3}, {name: "memory", value: 4, quantity: "300Mi"}]
+//
+// Version Comparisons:
+//
+// versionIsGreaterThan
+//
+// Returns true if the first version string is greater than the second version string.
+// The version must follow Semantic Versioning specification (http://semver.org/).
+// It can be with or without 'v' prefix (eg, "1.14.3" or "v1.14.3").
+//
+//	versionIsGreaterThan(<string>, <string>) <bool>
+//
+// Examples:
+//
+//	versionIsGreaterThan("1.25.0", "1.24.0") // returns true
+//	versionIsGreaterThan("1.24.0", "1.25.0") // returns false
+//
+// versionIsLessThan
+//
+// Returns true if the first version string is less than the second version string.
+// The version must follow Semantic Versioning specification (http://semver.org/).
+// It can be with or without 'v' prefix (eg, "1.14.3" or "v1.14.3").
+//
+//	versionIsLessThan(<string>, <string>) <bool>
+//
+// Examples:
+//
+//	versionIsLessThan("1.24.0", "1.25.0") // returns true
+//	versionIsLessThan("1.25.0", "1.24.0") // returns false
+//
+// Quantity Comparisons:
+//
+// quantityIsGreaterThan
+//
+// Returns true if the first quantity string is greater than the second quantity string.
+//
+//	quantityIsGreaterThan(<string>, <string>) <bool>
+//
+// Examples:
+//
+//	quantityIsGreaterThan("2Gi", "1Gi") // returns true
+//	quantityIsGreaterThan("1Gi", "2Gi") // returns false
+//	quantityIsGreaterThan("1000Mi", "1Gi") // returns false
+//
+// quantityIsLessThan
+//
+// Returns true if the first quantity string is less than the second quantity string.
+//
+//	quantityIsLessThan(<string>, <string>) <bool>
+//
+// Examples:
+//
+//	quantityIsLessThan("1Gi", "2Gi") // returns true
+//	quantityIsLessThan("2Gi", "1Gi") // returns false
+//	quantityIsLessThan("1000Mi", "1Gi") // returns true
```

Comment:
Can https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiserver/pkg/cel/library/quantity.go be leveraged here?  It provides comparisons.

Reply:
yes, I think so. 
The previous concern it seems the expression will be a very long and complex string if use k8s library.  For example: 
```
managedCluster.scores("default").filter(s, s.name == 'memAvailable').all(e, e.quantity.quantityIsGreaterThan("100Mi"))
```
vs
```
managedCluster.scores("default").filter(s, s.name == 'memAvailable').all(e, quantity(e.quantity).isGreaterThan(quantity("100Mi"))
```
But when I'm writing to compare the two expressions today, it seems no big difference, both are fine. We can try to use the k8s library instead.

Reply:
I've opened https://github.com/kubernetes/kubernetes/issues/130639 to track one small improvement that can be made here.

+100 to using the k8s library instead of forking off and starting a nearly identical one.